src/foam/nanos/alarming/AlarmConfigName.js

```diff
@@ -27,13 +26,24 @@ foam.CLASS({
   methods: [
     {
       name: 'put',
+      args: [
+        {
+          name: 'configName',
+          type: 'String'
+        }
+      ],
       javaCode: `
-        getDelegate().put(obj, sub);
-
-        String name = (String) obj;
-        DAO alarmConfigDAO = (DAO) getX().get("AlarmConfigDAO");
-        AlarmConfig config = (AlarmConfig) alarmConfigDAO.find(EQ(AlarmConfig.NAME, name));
-        if ( config == null || ! config.getEnabled() ) {
+        DAO alarmConfigDAO = (DAO) getX().get("alarmConfigDAO");
```

Comment:
Does this need to be synchronized to avoid adding twice by multiple threads? Is that possible?

Reply:
Potentially, but the 'name' is the ID, so the same would be added twice - create, update.